translate/convert/po2l20n.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2002-2015 Zuza Software Foundation
+#
+# This file is part of translate.
+#
+# translate is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# translate is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>.
+
+"""Convert Gettext PO localization files to .l20n files.
+"""
+
+import logging
+import re
+import warnings
+
+from translate.convert import accesskey, convert
+from translate.lang import data
+from translate.misc import quote
+from translate.storage import po, l20n
+
+
+logger = logging.getLogger(__name__)
+
+
+class po2l20n:
+
+    def __init__(self, includefuzzy=False):
+        self.includefuzzy = includefuzzy
+
+    def convertunit(self, unit):
+        l20n_unit = l20n.l20nunit(unit.getlocations()[0])
+        l20n_unit.value = unit.target or unit.source
+        #l20n_unit.comment = unit.allcomments
+
+        return l20n_unit
+
+    def mergestore(self, origl20nfile, translatedl20nfile):
+        target_store = l20n.l20nfile()
+        translatedl20nfile.makeindex()
+        for l20nunit in origl20nfile.units:
+            l20nunit_id = l20nunit.getid()
+            if l20nunit_id in translatedl20nfile.id_index:
+                target_store.addunit(translatedl20nfile.id_index[l20nunit_id])
+
+        return target_store
+
+    def convertstore(self, po_store):
+        outputstore = l20n.l20nfile()
+        for unit in po_store.units[1:]:
+            if not self.includefuzzy and unit.isfuzzy():
+                continue
+            newunit = self.convertunit(unit)
+            if newunit is not None:
+                outputstore.addunit(newunit)
+        return outputstore
+
+
+def convertl20n(inputfile, outputfile, templatefile, includefuzzy=False,
+                outputthreshold=None):
+    inputstore = po.pofile(inputfile)
+    convertor = po2l20n(includefuzzy)
+    outputl20n = convertor.convertstore(inputstore)
+    if templatefile is not None:
+        templatestore = l20n.l20nfile(templatefile)
+        outputl20n = convertor.mergestore(templatestore, outputl20n)
```

Comment:
Ideally this would be accepting a template l20n file and a translated PO file. I suggest adjusting the code in order to do that since it will simplify refactor in the future.